### PR TITLE
log the unresolved CDI devices only

### PR DESCRIPTION
**Pull request by tariq1890**
In `InjectDevices()` we are currently error-logging the whole list of devices instead of the device handles which weren't resolved

### CODE REVIEW

**Comment by tariq1890** [comment]
And some bonus changes to get the CI to play nice with me

**Review by elezar — APPROVED** [review]
Thanks @tariq1890.

One note. Is there a test that could be updated for this?

Not a blocker though.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 *.swp
 schema/validate
 /bin
+vendor/
```
Comment (klihub) [inline .gitignore:4]: nitpick: not a blocker necessarily, but this looks like an unrelated (and perhaps also unnecessary) change.

**Review by klihub — APPROVED** [review]
LGTM

File: .gitignore (hunk 1 above)
Comment (elezar) [reply .gitignore:4]: Good point. @tariq1890 do you mind rolling back this change and the one in the sanity workflow?

Feel free to create a separate PR for these though.

Comment (tariq1890) [reply .gitignore:4]: Sure I can remove this change, but I think we need the change in the sanity workflow to get the CI checks to pass

File: .github/workflows/sanity.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
       with:
         go-version: ${{ env.GO_VERSION }}
     - name: Install golint
-      run: go get -u golang.org/x/lint/golint
+      run: go install golang.org/x/lint/golint@latest
```
Comment (tariq1890) [inline .github/workflows/sanity.yml:25]: @elezar I was not able to get the CI checks to pass without this. I believe this is needed as a `go get` alters the go mod state

**Comment by tariq1890** [comment]
@klihub would you be able to review this?

**Review by klihub — APPROVED** [review]
LGTM.
